### DISCUSSION TITLE
Seurat V5 on MERFISH data

### DISCUSSION THREAD

**Discussion by YidaZhang0628** [satijalab/seurat#7170] [Q&A]
I am planning to use Seurat V5 on a MERFISH dataset. The tutorial uses `LoadVizgen` function to read the files. However, I don't have hdf5r files from segmentation. Therefore, I won't be able to use `LoadVizgen`. I can read the data using `ReadVizgen` but it results in a plain list instead of a Seurat object. Any idea on how to read MERFISH data without segmentation information into a Seurat object?

**Comment by resc0034** [comment]
I have the same question. It seems output from Vizgen MERSCOPE experiments will no longer provide a cell_boundaries folder with .hdf5 files. Instead, the output will include a "cell_boundaries.parquet" file. Are there any plans to update the LoadVizgen() function for use with a parquet file?

**Comment by alikhuseynov** [comment]
It seems that you have Cellpose output. I submitted a PR handle Cellpose output #7190 
See also this #7080
You may it `devtools::install_github(repo = "alikhuseynov/seurat", ref = "feat/vizgen")`
To load Vizgen this workflow could help [make Seurat objects](https://github.com/alikhuseynov/add-on_R/blob/develop/R/example_read_load_vizgen_v2.R)

**Comment by MaycieS** [comment]
Hello. I am having issues using LoadVizgen() and ReadVizgen() with my MERFISH dataset still. I know there was a PR made for this, but I am still confused on how to fix the parquet issue. When I use the code below,


vizgen.obj <- ReadVizgen(data.dir = "E:\\111124_MS_MB_Merfishy", transcripts = "E:\\111124_MS_MB_Merfishy\\cell_by_gene.csv", spatial = "E:\\111124_MS_MB_Merfishy\\cell_metadata.csv", molecules = "E:\\111124_MS_MB_Merfishy\\detected_transcripts.csv", type = "segmentations", mol.type = "microns", z=3L)

I get the error "Warning in ReadVizgen(data.dir = data_dir, transcripts = transcripts_path,  :
  180046 cells missing polygon information"

There were 50 or more warnings (use warnings() to see the first 50)". I am assuming that is because there is no hfad file anymore (replaced by parquet). What code am I supposed to run instead?

**Reply by alikhuseynov** [reply]
Hi. try to install this PR for Seurat5 `devtools::install_github(repo = "alikhuseynov/seurat", ref = "vizgen_seurat5")`
try to use [this loading script](https://github.com/alikhuseynov/add-on_R/blob/develop/R/example_read_load_vizgen_v2.R)
> vizgen.obj <- ReadVizgen(data.dir = "E:\111124_MS_MB_Merfishy", transcripts = "E:\111124_MS_MB_Merfishy\cell_by_gene.csv", spatial = "E:\111124_MS_MB_Merfishy\cell_metadata.csv", molecules = "E:\111124_MS_MB_Merfishy\detected_transcripts.csv", type = "segmentations", mol.type = "microns", z=3L)

`ReadVizgen` doesn't make the object, it reads the data in a list, `LoadVizgen` needs to be used. Make sure that `parquet` file is somewhere/within your `E:\111124_MS_MB_Merfishy` directory.
hope this helps.

**Comment by MaycieS** [comment]
Thank you! That appeared to work, however I now cannot run the

vizgen.obj <- SCTransform(vizgen.obj, assay = "Vizgen") function due to the class = list.

I know I need this to be a Seurat Object in matrix form. Is there a way to fix that?


Thanks,

Maycie


Maycie Schultz
Sun Lab
Biomedical Sciences Graduate Student
University of Wyoming, Biological Sciences 1000 E University Ave, Laramie, WY 82071
________________________________

**Reply by alikhuseynov** [reply]
ok, no worries. 
You need `vizgen.obj` to be a Seurat object not an output `list` class after `ReadVizgen`. `LoadVizgen` returns Seurat object and that should work with `SCTransform`.
